tests/drivers/pinctrl/nrf/boards/nrf52840dk_nrf52840.overlay

```diff
@@ -0,0 +1,42 @@
+/*
```

Comment:
Why bother limiting this to nRF52840-DK?

It's already in an nrf directory, and this should work on all SoCs, so you could just make this app.overlay and allow testing it on any board a person has available -- importantly, including nrf52_bsim.

Reply:
moved to app.overlay. I'm still filtering with `platform_allow`, though. I'm not sure what would be the best way to allow this test to run on all nRF based targets. Filtering by Kconfig is not an option, since config step would fail due to the app.overlay content.

Reply:
Can't you use integration_platforms to specify this should be built with board 'foo', without limiting to just 'foo'? IIRC that's the difference between platform_allow and integration_platforms.

Reply:
good point, I have updated to `integration_platforms`.

Reply:
@mbolivar-nordic CI tries to build for `native_posix` now (https://github.com/zephyrproject-rtos/zephyr/runs/4190937903?check_suite_focus=true), suggestions?

Reply:
I'm out of my element, that's not how I thought twister was supposed to work. @PerMac can you help us restrict the twister run to just the platforms in integration_platforms for this PR?

@gmarull maybe `filter: dt_compat_enabled("nordic,nrf-pinctrl")` will help?

Reply:
It will not work, filter relies on the DT pickle file generated at configuration time, a step that will fail (due to DT syntax errors, `NRF_PIN` not available).

Reply:
A solution may be to rely on tags

Reply:
CONFIG_SOC_COMPATIBLE_NRF?

Reply:
same problem: it relies on CMake config step to succeed.

Reply:
I suggest adding `platform_exclude: native_posix` to satisfy the CI. The gh action that failed (clang-build) has native_posix explicitly given in the twister call there. It does not use `integration_platforms` nor `--integration` (integration mode). `Integration_platforms` are used only when `--integration` is used. 
I know that platform selection in CI is hard to follow but until we upgrade twister there probably is no better way for this. 
As @gmarull pointed out: we can only use 'yaml parsing' ways to filter in this case. Using dt or kconfig filtering is only possible after initial cmake passes.